src/validate-and-json.js

```diff
@@ -34,47 +33,48 @@ const reportBrokenEdge = (
                 sourceFile,
                 options.rootMarker,
             )}`,
-        });
-        return;
+        };
     }
 
     const NO_CHECKSUM = "No checksum";
     const sourceFileRef = rootRelativePath(
         `${sourceFile}:${sourceLine}`,
         options.rootMarker,
     );
-    jsonItems.push({
+
+    return {
         type: "violation",
         sourceFile: rootRelativePath(sourceFile, options.rootMarker),
-        sourceLine: sourceLine,
+        sourceLine: parseInt(sourceLine, 10),
         targetFile: rootRelativePath(targetFile, options.rootMarker),
-        targetLine: targetLine,
+        targetLine: parseInt(targetLine, 10),
```

Comment:
Hmm, interesting. I can't imagine a circumstance where this would fail.

Reply:
It might make sense to move the parsing further up so that the line numbers are a numbers throughout the majority of the code.